src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -72,15 +72,26 @@ struct InternalFlashFactoryData
 
     constexpr size_t FactoryDataBlockSize()
     {
-        // calculate the factory data end address rounded up to the nearest multiple of CONFIG_FPROTECT_BLOCK_SIZE
+        // calculate the factory data and settings end address rounded up to the nearest multiple of CONFIG_FPROTECT_BLOCK_SIZE
         // and make sure we do not overlap with settings partition
         constexpr size_t kFactoryDataBlockEnd =
             (FACTORY_DATA_ADDRESS + FACTORY_DATA_SIZE + CONFIG_FPROTECT_BLOCK_SIZE - 1) & (-CONFIG_FPROTECT_BLOCK_SIZE);
-        static_assert(kFactoryDataBlockEnd <= PM_SETTINGS_STORAGE_ADDRESS,
+
+        constexpr size_t SettingsBlockEnd =
```

Comment:
@ArekBalysNordic please correct me if I'm wrong, but I think that the settings partition does not need to be aligned to the fprotect block size, so the end address is just start address + size, right?

Reply:
That's right. Only the partition that is protected by fprotect must be aligned to fprotect block size. So we need to take care of factory data only. We need to be aware to not overlap settings partition by fprotect. Keep in mind that the project block size varies depending on our platform.